### PR TITLE
Misc Improvements and Bug Fixes

### PR DESCRIPTION
**Pull request by AndrewKahr**
#### Changes

- Add tests for Windows images
- Consolidate testing workflows
- Fix checking out main instead of latest tag for building images
- Fix using GITHUB_ENV rather than GITHUB_OUTPUT which broke Windows image builds
- Improve error checking on Windows editor installations
- General improvements to error checking on editor/module installations
- Hide node error spam from Unity Hub by piping stderr to null. This is ok because the errors we care about end up on stdout

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)

### CODE REVIEW

File: images/ubuntu/hub/Dockerfile [hunk 1]
```diff
@@ -16,8 +16,11 @@ RUN sh -c 'echo "deb https://hub.unity3d.com/linux/repos/deb stable main" > /etc
  && apt-get -q install -y "unityhub=$hubVersion" \
  && apt-get clean
 
+# Bug in xvfb-run that causes it to redirect stderr to stdout. We want it separate
```
Comment (GabLeRoux) [inline images/ubuntu/hub/Dockerfile:19]: Do you have details about that bug?

Comment (AndrewKahr) [reply images/ubuntu/hub/Dockerfile:19]: https://bugs.launchpad.net/ubuntu/+source/xorg-server/+bug/1059947
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=868876

**Review by GabLeRoux — COMMENTED** [review]
I think it would be worth it at some point to extract some scripts from some of the workflow files for readability (ex; in `.github/workflows/test.yml`). Other than that, this looks good to me 👍 🤯

**Review by GabLeRoux — APPROVED** [review]

**Comment by GabLeRoux** [comment]
The following two jobs both failed with a timeout, but everything else is green.

https://github.com/game-ci/docker/actions/runs/4675713024/jobs/8286515438?pr=206
https://github.com/game-ci/docker/actions/runs/4675713024/jobs/8286515018?pr=206

Are we ok with that?

**Comment by AndrewKahr** [comment]
I checked the timeouts and they are from Unity not exiting despite completing the build successfully in well under 15 minutes. This is a known issue but out of scope for this PR so it is acceptable. The images themselves are building and installing the correct modules which is the main thing that matters for the scope of this PR.

**Pull request merged**